llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2653,6 +2657,260 @@ class VPReductionEVLRecipe : public VPReductionRecipe {
   }
 };
 
+/// A recipe to represent inloop extended reduction operations, performing a
+/// reduction on a vector operand into a scalar value, and adding the result to
+/// a chain. This recipe is high level abstract which will generate
+/// VPReductionRecipe and VPWidenCastRecipe before execution. The Operands are
+/// {ChainOp, VecOp, [Condition]}.
+class VPExtendedReductionRecipe : public VPSingleDefRecipe {
```

Comment:
How much extra code would be needed to add the required info to the existing VPReductionRecipe? If to much, could we at least avoid most duplication in the new recipe by inheriting from VPReductionRecipe?

Reply:
Inherited from VPReductionRecipe, thanks.